models/baseModels/Item/Item.ts

```diff
@@ -18,6 +18,34 @@ export class Item extends Doc {
   trackItem?: boolean;
   itemType?: 'Product' | 'Service';
   for?: 'Purchases' | 'Sales' | 'Both';
+  hasBatchNumber?: boolean;
+
+  async beforeSync() {
```

Comment:
This code block should not be here. It should be in the `validations` property that checks if `hasBatchNumber` is being unset.

Either that or mark it as a readOnly like the trackItem field.

In fact do the latter, there's no need for unnecessary validations.

Reply:
corrected